MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -42,6 +41,53 @@ final class MapViewModel: ObservableObject {
         CLLocationManager().requestWhenInUseAuthorization()
         CLLocationManager().desiredAccuracy = kCLLocationAccuracyBest
     }
+
+    func onNearbyFarmerButtonTapped() {
+        do {
+            if let nearbyFarmer = try searchNearbyFarmer() {
+                let nearbyFarmerCoordinate = CLLocationCoordinate2D(latitude: nearbyFarmer.location.latitude,
+                                                                    longitude: nearbyFarmer.location.longitude)
+                let nearbyFarmerRegion = MKCoordinateRegion(center: nearbyFarmerCoordinate,
+                                                            span: self.nearbyFarmerSpan)
+                mapCameraPosition = .region(nearbyFarmerRegion)
+            }
+        } catch {
+            print("Aucun maraîcher trouvé à proximité")
+        }
+    }
+
+    private func searchNearbyFarmer() throws -> Farmer? {
+        var maxSearchDistance = 10_000.0
+        var nearbyFarmer: Farmer?
+        guard let userLocation = CLLocationManager().location else { return nil }
+        for farmer in allFarmers {
+            let distance = userLocation.distance(from: .init(latitude: farmer.location.latitude, longitude: farmer.location.longitude))
+            if distance < maxSearchDistance {
+                maxSearchDistance = distance
+                nearbyFarmer = farmer
+            }
+        }
+        return nearbyFarmer
+    }
+}
+
+extension MapViewModel {
+
+    enum Error: Swift.Error {
```

Comment:
Are you using it?

Reply:
I would like to. In the event that no farmer was found nearby

Reply:
Is this use case an error? Or just an information that there is no farmer close to the user?

Reply:
You’re right, it’s not a mistake but information for the user